compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5209,6 +5235,88 @@ private boolean isUnaryOperator(SyntaxKind kind) {
         }
     }
 
+    /**
+     * Parse array type descriptor.
+     * <p>
+     * <code>
+     * array-type-descriptor := member-type-descriptor [ [ array-length ] ]
+     * member-type-descriptor := type-descriptor
+     * array-length :=
+     *    int-literal
+     *    | constant-reference-expr
+     *    | inferred-array-length
+     * inferred-array-length := *
+     * </code>
+     * </p>
+     * @param typeDescriptorNode
+     * @return Parsed Node
+     */
+
+    private STNode parseArrayTypeDescriptor(STNode typeDescriptorNode) {
+        startContext(ParserRuleContext.ARRAY_TYPE_DESCRIPTOR);
+        STNode dimensions = parseDimensions();
+
+        endContext();
+        return STNodeFactory.createArrayTypeDescriptor(typeDescriptorNode, dimensions);
```

Comment:
I think the structure of the resulting tree is a bit different to the spec. 
e.g. for something like int[][][], this model group the tree nodes as `(int) ([][][])`.
But in the spec, it is: `(((int)[])[])[]`.

I feel recursion would solve this.

Reply:
I'll fix that.

Reply:
@SupunS I have fixed this. Please review.